### ISSUE TITLE
Compilation error when using `effectful-plugin` without `effectful-core`

### ISSUE THREAD

**Issue by MuhammedZakir** [haskell-effectful/effectful#86]
The error I am getting:

```
ghc: panic! (the 'impossible' happened)
  (GHC version 9.2.2:
        Unable to resolve module looked up by plugin:
  Effectful.Internal.Effect

Please report this as a GHC bug:  https://www.haskell.org/ghc/reportabug
```

This is resolved if I also add `effectful-core` to dependencies.

I am reporting here in case `effectful-plugin` should work with just `effectful`.

---

Configs when I got that error (i.e. before adding `effectful-core`):

`dependencies` in `package.yaml`:
```yaml
dependencies:
- base >= 4.7 && < 5
- effectful
- effectful-plugin
- effectful-th
- resourcet-effectful
- unliftio

language: GHC2021

default-extensions:
- BlockArguments
- GADTs
- LambdaCase
- OverloadedStrings
- ScopedTypeVariables
- TemplateHaskell
- TypeFamilies

ghc-options:
- -fplugin=Effectful.Plugin
# default options in the template
- -Wall
- -Wcompat
- -Widentities
- -Wincomplete-record-updates
- -Wincomplete-uni-patterns
- -Wmissing-export-lists
- -Wmissing-home-modules
- -Wpartial-fields
- -Wredundant-constraints
```

`stack.yaml`:
```yaml
resolver: nightly-2022-06-22
compiler: ghc-9.2.2
allow-newer: true

packages:
- .

extra-deps:
- effectful-2.0.0.0
- effectful-core-2.0.0.0
- effectful-plugin-1.0.0.0
- effectful-th-1.0.0.0
- resourcet-effectful-1.0.0.0

pvp-bounds: both
```

**Comment by arybczak** [comment]
This is fixed in effectful-plugin-1.1.0.0, unfortunately only for GHC >= 9.4.

**Comment by arybczak** [comment]
Turns out it's not actually fixed. I think the best that can be done is to show a better error message, which was done in 7a23dc47692c428eadd3306af1a61c4d158ba9ac.
